File: src/main/scala/s/template_method/delegate/CharDisplayDelegate.scala
package s.template_method.delegate

final class CharDisplayDelegate(private val ch: Char) extends DisplayDelegate {
  override def open(): Unit = {
    Console.print("<<")
  }

  override def print(): Unit = {
    Console.print(ch)
  }

  override def close(): Unit = {
    Console.print(ch)
  }
}

File: src/main/java/j/template_method/delegate/CharDisplayDelegate.java
package j.template_method.delegate;

public final class CharDisplayDelegate implements DisplayDelegate {
  private final char ch;

  CharDisplayDelegate(char ch) {
    this.ch = ch;
  }

  @Override
  public void open() {
    System.out.print("<<");
  }

  @Override
  public void print() {
    System.out.print(ch);
  }

  @Override
  public void close() {
    System.out.print(ch);
  }
}
